include/swift/SILOptimizer/Differentiation/ADContext.h

```diff
@@ -269,6 +272,10 @@ class ADContext {
                                 Diag<T...> diag, U &&... args);
 };
 
+raw_ostream &getADDebugStream();
+SILLocation getValidLocation(SILValue v);
```

Comment:
Is there a more descriptive name than get_Valid_Location() we could use here? Or perhaps add a Doxygen comment explaining what this function does?